File: JavaScript/Programmers_Solving/Level1/명예의 전당(1).js
const k = 3;
const score = [10, 100, 20, 150, 1, 100, 200];

function solution(k, score) {
  var answer = [];
  let awardStack = [];
  for (i = 0; i < score.length; i++) {
    awardStack.push(score[i]);
    awardStack.sort((a, b) => b - a);
    if (k < awardStack.length) {
      awardStack.pop();
    }
    answer.push(awardStack[awardStack.length - 1]);
  }
  return answer;
}

console.log(solution(k, score));

File: Python/Programmers_Solving/Level1/명예의 전당(1).py
def solution(k, score):
    answer = []
    awardStack = []
    for i in score :
        awardStack.append(i)
        awardStack.sort(reverse=True)
        if(len(awardStack) > k) :
            awardStack.pop()
        answer.append(awardStack[-1])
    return answer